pephub/routers/api/v1/project.py

```diff
@@ -308,3 +318,43 @@ async def convert_pep(
 async def zip_pep_for_download(proj: peppy.Project = Depends(get_project)):
     """Zip a pep"""
     return zip_pep(proj)
+
+
+@project.post("/forks",
+              summary="Fork project to user namespace.",
+              )
+async def fork_pep_to_namespace(
+    project: str,
+    fork_namespace: Annotated[str, Form()] = Form(),
+    tag: str = DEFAULT_TAG,
+    namespace_access_list: List[str] = Depends(get_namespace_access_list),
+    proj: peppy.Project = Depends(get_project),
+    agent: PEPDatabaseAgent = Depends(get_db),
+):
+    if fork_namespace in (namespace_access_list or []):
+        try:
+            agent.project.create(project=proj, namespace=fork_namespace, name=project, tag=tag)
+        except ProjectUniqueNameError as e:
+            return JSONResponse(
+                content={
+                    "message": f"Project '{fork_namespace}/{project}:{tag}' already exists in namespace",
+                    "error": f"{e}",
+                    "status_code": 400,
+                },
+                status_code=400,
+            )
+
+        return JSONResponse(
+            content={
+                "namespace": fork_namespace,
+                "project_name": project,
+                "tag": tag,
+                "registry_path": f"{fork_namespace}/{project}:{tag}",
+            },
+            status_code=202,
+        )
+    else:
```

Comment:
Should we add a `user_can_fork` dependency that checks this for us? I like what you added, and I think it will be helpful in the future.

We can add it here:
```python
@project.post("/forks", summary="Fork project to user namespace.")
```

Reply:
I think we shouldn't `user_can_fork` endpoint. If user can read project, then it means user can fork it